--- a/configure.ac	2018-02-04 20:32:18.639142401 +0100
+++ b/configure.ac	2018-02-04 20:32:59.019469285 +0100
@@ -337,6 +337,7 @@
         [], [], [[
 #include <sys/types.h>
 #include <unistd.h>
+#include <sys/mman.h>
 #include <sys/mount.h>
 #include <fcntl.h>
 #include <sched.h>
--- a/src/basic/fileio.c	2018-02-04 20:33:47.676388500 +0100
+++ b/src/basic/fileio.c	2018-02-04 20:33:51.309735208 +0100
@@ -24,6 +24,7 @@
 #include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
+#include <sys/mman.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <unistd.h>
