--- a/libretro/Makefile.libretro	2019-03-05 15:01:28.262047406 +0100
+++ b/libretro/Makefile.libretro	2019-03-05 14:58:57.113138085 +0100
@@ -183,7 +183,7 @@
 	COMMONFLAGS += -DC_HAVE_MPROTECT="1"
 	TARGET := $(TARGET_NAME)_libretro.so
 	LDFLAGS += -shared -Wl,--version-script=link.T
-	LDFLAGS += `sdl-config --static-libs` -lSDL_net
+	LDFLAGS += `${SYSROOT_PREFIX}/usr/bin/sdl-config --static-libs` -lSDL_net
 	COMMONFLAGS += -DUNIX_PORTABLE
 	LDFLAGS += -Wl,-Bdynamic
 	fpic = -fPIC
