--- a/libretro/Makefile.common	2018-11-19 23:55:48.602994863 +0100
+++ b/libretro/Makefile.common	2018-11-19 23:58:40.767273666 +0100
@@ -9,6 +9,7 @@
 	-I$(CORE_DIR)/src \
 	-I$(CORE_DIR)/libretro \
 	-I$(LIBRETRO_COMM_DIR)/include \
+	-I$(LIBRETRO_SDL_DIR)/include \
 	-I$(CORE_DIR)/include \
 	-I$(CORE_DIR)/src/midi/munt
 
--- a/libretro/Makefile.libretro	2019-03-05 15:01:28.262047406 +0100
+++ b/libretro/Makefile.libretro	2019-03-05 14:58:57.113138085 +0100
@@ -183,7 +183,7 @@
 	COMMONFLAGS += -DC_HAVE_MPROTECT="1"
 	TARGET := $(TARGET_NAME)_libretro.so
 	LDFLAGS += -shared -Wl,--version-script=link.T
-	LDFLAGS += `sdl-config --static-libs` -lSDL_net
+	LDFLAGS += `${SYSROOT_PREFIX}/usr/bin/sdl-config --static-libs` -lSDL_net
 	COMMONFLAGS += -DUNIX_PORTABLE
 	LDFLAGS += -Wl,-Bdynamic
 	fpic = -fPIC
