--- a/qb/qb.params.sh	2019-03-24 20:30:37.544264426 +0100
+++ b/qb/qb.params.sh	2019-03-24 20:30:43.624138987 +0100
@@ -62,7 +62,6 @@
 	err="$2"
 	eval "set -- $OPTS"
 	for OPT do [ "$opt" = "$OPT" ] && return; done
-	die 1 "Unknown option $err"
 }
 
 parse_input() # Parse stuff :V
@@ -106,7 +105,6 @@
 			-h|--help) print_help; exit 0;;
 			--) break ;;
 			'') : ;;
-			*) die 1 "Unknown option $1";;
 		esac
 		shift
 	done
