--- a/wifi/wifi_driver.c	2019-04-05 15:26:57.452496407 +0200
+++ b/wifi/wifi_driver.c	2019-04-05 15:27:17.552230110 +0200
@@ -34,9 +34,7 @@
 static bool wifi_driver_active              = false;
 
 static const wifi_driver_t *wifi_drivers[] = {
-#ifdef HAVE_LAKKA
    &wifi_connmanctl,
-#endif
    &wifi_null,
    NULL,
 };
--- a/Makefile.common	2019-04-05 15:35:15.756056497 +0200
+++ b/Makefile.common	2019-04-05 15:35:25.889260350 +0200
@@ -511,9 +511,7 @@
           camera/drivers/rwebcam.o
 endif
 
-ifeq ($(HAVE_LAKKA), 1)
    OBJ += wifi/drivers/connmanctl.o
-endif
 
 # Audio
 
--- a/menu/menu_setting.c	2019-04-09 21:14:05.838796905 +0200
+++ b/menu/menu_setting.c	2019-04-09 21:14:44.658275483 +0200
@@ -4582,7 +4582,6 @@
                parent_group);
          settings_data_list_current_add_flags(list, list_info, SD_FLAG_ADVANCED);
 
-#ifdef HAVE_LAKKA
          if (string_is_not_equal(settings->arrays.wifi_driver, "null"))
          {
             CONFIG_ACTION(
@@ -4593,7 +4592,6 @@
                   &subgroup_info,
                   parent_group);
          }
-#endif
 
          CONFIG_ACTION(
                list, list_info,
