--- a/arch/arm/boot/dts/exynos5420.dtsi	2019-08-25 21:01:23.000000000 +0200
+++ b/arch/arm/boot/dts/exynos5420.dtsi	2019-08-31 15:05:53.720153683 +0200
@@ -41,11 +41,21 @@
 	 * The 'cpus' node is not present here but instead it is provided
 	 * by exynos5420-cpus.dtsi or exynos5422-cpus.dtsi.
 	 */
-
+  soc: soc {
 	cluster_a15_opp_table: opp_table0 {
 		compatible = "operating-points-v2";
 		opp-shared;
 
+		opp-2000000000 {
+			opp-hz = /bits/ 64 <2000000000>;
+			opp-microvolt = <1312500>;
+			clock-latency-ns = <140000>;
+		};
+		opp-1900000000 {
+			opp-hz = /bits/ 64 <1900000000>;
+			opp-microvolt = <1250000>;
+			clock-latency-ns = <140000>;
+		};
 		opp-1800000000 {
 			opp-hz = /bits/ 64 <1800000000>;
 			opp-microvolt = <1250000>;
@@ -112,6 +122,16 @@
 		compatible = "operating-points-v2";
 		opp-shared;
 
+		opp-1500000000 {
+			opp-hz = /bits/ 64 <150000000>;
+			opp-microvolt = <1312500>;
+			clock-latency-ns = <140000>;
+		};
+		opp-1400000000 {
+			opp-hz = /bits/ 64 <140000000>;
+			opp-microvolt = <1250000>;
+			clock-latency-ns = <140000>;
+		};
 		opp-1300000000 {
 			opp-hz = /bits/ 64 <1300000000>;
 			opp-microvolt = <1275000>;
@@ -154,7 +174,7 @@
 		};
 	};
 
-	soc: soc {
+	
 		cci: cci@10d20000 {
 			compatible = "arm,cci-400";
 			#address-cells = <1>;
@@ -1328,7 +1348,60 @@
 			};
 		};
 	};
+	mali: mali@11800000 {
+		compatible = "samsung,exynos5422-mali", "arm,malit628", "arm,malit62x", "arm,malit6xx", "arm,mali-midgard";
+		reg = <0x11800000 0x5000>;
+		interrupts =  <GIC_SPI 219 IRQ_TYPE_LEVEL_HIGH>, <GIC_SPI 74 IRQ_TYPE_LEVEL_HIGH>, <GIC_SPI 117 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-names = "JOB", "MMU", "GPU";
+
+		clocks = <&clock CLK_FOUT_VPLL>, <&clock CLK_DOUT_ACLK_G3D>, <&clock CLK_G3D>;
+		clock-names = "fout_vpll", "dout_aclk_g3d", "clk_mali";
+
+		operating-points-v2 = <&gpu_opp_table>;
+
+		status = "disabled";
+
+		power_model@0 {
+			compatible = "arm,mali-simple-power-model";
+			static-coefficient = <2427750>;
+			dynamic-coefficient = <4687>;
+			ts = <20000 2000 (-20) 2>;
+			thermal-zone = "gpu-thermal";
+		};
+	};
+
+	gpu_opp_table: opp_table0 {
+		compatible = "operating-points-v2";
 
+		opp@600000000 {
+			opp-hz = /bits/ 64 <600000000>;
+			opp-microvolt = <1150000>;
+		};
+		opp@543000000 {
+			opp-hz = /bits/ 64 <543000000>;
+			opp-microvolt = <1037500>;
+		};
+		opp@480000000 {
+			opp-hz = /bits/ 64 <480000000>;
+			opp-microvolt = <1000000>;
+		};
+		opp@420000000 {
+			opp-hz = /bits/ 64 <420000000>;
+			opp-microvolt = <962500>;
+		};
+		opp@350000000 {
+			opp-hz = /bits/ 64 <350000000>;
+			opp-microvolt = <912500>;
+		};
+		opp@266000000 {
+			opp-hz = /bits/ 64 <266000000>;
+			opp-microvolt = <862500>;
+		};
+		opp@177000000 {
+			opp-hz = /bits/ 64 <177000000>;
+			opp-microvolt = <812500>;
+		};
+	};																										 
 	thermal-zones {
 		cpu0_thermal: cpu0-thermal {
 			thermal-sensors = <&tmu_cpu0>;
--- a/arch/arm/boot/dts/exynos5422-cpus.dtsi	2019-08-31 12:29:37.479914349 +0200
+++ b/arch/arm/boot/dts/exynos5422-cpus.dtsi	2019-08-31 12:31:39.278273699 +0200
@@ -26,7 +26,7 @@
 			compatible = "arm,cortex-a7";
 			reg = <0x100>;
 			clocks = <&clock CLK_KFC_CLK>;
-			clock-frequency = <1000000000>;
+			clock-frequency = <1500000000>;
 			cci-control-port = <&cci_control0>;
 			operating-points-v2 = <&cluster_a7_opp_table>;
 			#cooling-cells = <2>; /* min followed by max */
@@ -38,7 +38,7 @@
 			compatible = "arm,cortex-a7";
 			reg = <0x101>;
 			clocks = <&clock CLK_KFC_CLK>;
-			clock-frequency = <1000000000>;
+			clock-frequency = <1500000000>;
 			cci-control-port = <&cci_control0>;
 			operating-points-v2 = <&cluster_a7_opp_table>;
 			#cooling-cells = <2>; /* min followed by max */
@@ -50,7 +50,7 @@
 			compatible = "arm,cortex-a7";
 			reg = <0x102>;
 			clocks = <&clock CLK_KFC_CLK>;
-			clock-frequency = <1000000000>;
+			clock-frequency = <1500000000>;
 			cci-control-port = <&cci_control0>;
 			operating-points-v2 = <&cluster_a7_opp_table>;
 			#cooling-cells = <2>; /* min followed by max */
@@ -62,7 +62,7 @@
 			compatible = "arm,cortex-a7";
 			reg = <0x103>;
 			clocks = <&clock CLK_KFC_CLK>;
-			clock-frequency = <1000000000>;
+			clock-frequency = <1500000000>;
 			cci-control-port = <&cci_control0>;
 			operating-points-v2 = <&cluster_a7_opp_table>;
 			#cooling-cells = <2>; /* min followed by max */
@@ -74,7 +74,7 @@
 			compatible = "arm,cortex-a15";
 			reg = <0x0>;
 			clocks = <&clock CLK_ARM_CLK>;
-			clock-frequency = <1800000000>;
+			clock-frequency = <2000000000>;
 			cci-control-port = <&cci_control1>;
 			operating-points-v2 = <&cluster_a15_opp_table>;
 			#cooling-cells = <2>; /* min followed by max */
@@ -86,7 +86,7 @@
 			compatible = "arm,cortex-a15";
 			reg = <0x1>;
 			clocks = <&clock CLK_ARM_CLK>;
-			clock-frequency = <1800000000>;
+			clock-frequency = <2000000000>;
 			cci-control-port = <&cci_control1>;
 			operating-points-v2 = <&cluster_a15_opp_table>;
 			#cooling-cells = <2>; /* min followed by max */
@@ -98,7 +98,7 @@
 			compatible = "arm,cortex-a15";
 			reg = <0x2>;
 			clocks = <&clock CLK_ARM_CLK>;
-			clock-frequency = <1800000000>;
+			clock-frequency = <2000000000>;
 			cci-control-port = <&cci_control1>;
 			operating-points-v2 = <&cluster_a15_opp_table>;
 			#cooling-cells = <2>; /* min followed by max */
@@ -110,7 +110,7 @@
 			compatible = "arm,cortex-a15";
 			reg = <0x3>;
 			clocks = <&clock CLK_ARM_CLK>;
-			clock-frequency = <1800000000>;
+			clock-frequency = <2000000000>;
 			cci-control-port = <&cci_control1>;
 			operating-points-v2 = <&cluster_a15_opp_table>;
 			#cooling-cells = <2>; /* min followed by max */
--- a/arch/arm/boot/dts/exynos5422-odroid-core.dtsi	2019-08-31 15:12:30.655017746 +0200
+++ b/arch/arm/boot/dts/exynos5422-odroid-core.dtsi	2019-08-31 14:52:16.253404224 +0200
@@ -675,3 +675,8 @@
 	vdd33-supply = <&ldo9_reg>;
 	vdd10-supply = <&ldo11_reg>;
 };
+
+&mali {	
+    mali-supply = <&buck4_reg>;	
+    status = "okay";
+};
