diff --git a/tools/l2test.c b/tools/l2test.c
index 1819423..7294d79 100644
--- a/tools/l2test.c
+++ b/tools/l2test.c
@@ -906,7 +906,7 @@ static void recv_mode(int sk)
 			if (timestamp) {
 				struct timeval tv;
 
-				if (ioctl(sk, SIOCGSTAMP, &tv) < 0) {
+				if (1) {
 					timestamp = 0;
 					memset(ts, 0, sizeof(ts));
 				} else {
diff --git a/tools/rctest.c b/tools/rctest.c
index 6d84e07..295c5aa 100644
--- a/tools/rctest.c
+++ b/tools/rctest.c
@@ -504,7 +504,7 @@ static void recv_mode(int sk)
 			if (timestamp) {
 				struct timeval tv;
 
-				if (ioctl(sk, SIOCGSTAMP, &tv) < 0) {
+				if (1) {
 					timestamp = 0;
 					memset(ts, 0, sizeof(ts));
 				} else {
