--- a/src/vicii/vicii-draw.c	2017-08-23 00:24:30.837140757 +0200
+++ b/src/vicii/vicii-draw.c	2017-08-23 00:24:40.460586587 +0200
@@ -162,7 +162,7 @@
    anyway.  */
 
 #ifndef ALLOW_UNALIGNED_ACCESS
-static uitn32_t _aligned_line_buffer[VICII_SCREEN_XPIX / 2 + 1];
+static uint32_t _aligned_line_buffer[VICII_SCREEN_XPIX / 2 + 1];
 static uint8_t *const aligned_line_buffer = (uint8_t *)_aligned_line_buffer;
 #endif
 
