--- a/libretro/libretro.c	2019-04-20 13:39:19.431941042 +0200
+++ b/libretro/libretro.c	2019-04-20 13:39:36.788416382 +0200
@@ -60,7 +60,6 @@
    { 0, RETRO_DEVICE_JOYPAD, 0, RETRO_DEVICE_ID_JOYPAD_RIGHT, "Right" },
    { 0, RETRO_DEVICE_JOYPAD, 0, RETRO_DEVICE_ID_JOYPAD_A, "Fire" },
    { 0, RETRO_DEVICE_JOYPAD, 0, RETRO_DEVICE_ID_JOYPAD_B, "Mouse btn B" },
-   { 0, RETRO_DEVICE_JOYPAD, 0, RETRO_DEVICE_ID_JOYPAD_Y, "Enter GUI" },
    { 0, RETRO_DEVICE_JOYPAD, 0, RETRO_DEVICE_ID_JOYPAD_SELECT, "Mouse mode toggle" },
    { 0, RETRO_DEVICE_JOYPAD, 0, RETRO_DEVICE_ID_JOYPAD_START, "Keyboard overlay" },
    { 0, RETRO_DEVICE_JOYPAD, 0, RETRO_DEVICE_ID_JOYPAD_L2, "Toggle m/k status" },
--- a/libretro/libretro-mapper.c	2019-04-20 13:39:26.531862820 +0200
+++ b/libretro/libretro-mapper.c	2019-04-20 13:39:58.394844393 +0200
@@ -237,7 +237,7 @@
    input_poll_cb();
    Process_key();
 
-   if (key_state[RETROK_F11] || input_state_cb(0, RETRO_DEVICE_JOYPAD, 0, RETRO_DEVICE_ID_JOYPAD_Y))
+   if (key_state[RETROK_F11])
    {
       pauseg=1;
       //enter_gui(); //old
