--- a/Makefile.common	2019-04-20 13:02:10.822516710 +0200
+++ b/Makefile.common	2019-04-20 12:53:23.566094807 +0200
@@ -25,6 +25,7 @@
 				  $(EMU)/inputrecord.c \
 				  $(EMU)/keymap/keymap.c\
 				  $(EMU)/diskutil.c \
+				  $(EMU)/caps/caps.c \
 				  $(EMU)/zfile.c \
 				  $(EMU)/zfile_archive.c \
 				  $(EMU)/cfgfile.c \
@@ -172,4 +173,4 @@
 				 $(DEPS_DIR)/zlib/zutil.c
 endif
 
-CFLAGS += -DLIBRETRO_PUAE
+CFLAGS += -DLIBRETRO_PUAE -DCAPS
--- a/sources/src/caps/caps.c	2019-04-20 13:02:31.148917617 +0200
+++ b/sources/src/caps/caps.c	2019-04-20 13:03:08.918423849 +0200
@@ -27,7 +27,7 @@
 
 #include <dlfcn.h>
 
-#define CAPSLIB_NAME    "libcapsimage.so.2"
+#define CAPSLIB_NAME    "capsimg.so"
 
 /*
  * Repository for function pointers to the CAPSLib routines
