--- a/libretro/Makefile.libretro	2019-04-02 14:20:04.000000000 +0200
+++ b/libretro/Makefile.libretro	2019-04-02 14:22:02.650925856 +0200
@@ -222,7 +222,7 @@
 		COMMONFLAGS += -DWITH_FAKE_SDL
 		LDFLAGS += -Wl,-Bstatic
 	else
-		LDFLAGS += `sdl-config --libs` -lSDL_net
+		LDFLAGS += `${SYSROOT_PREFIX}/usr/bin/sdl-config --static-libs` -lSDL_net
 	endif
 	COMMONFLAGS += -DUNIX_PORTABLE
 	LDFLAGS += -Wl,-Bdynamic
