--- a/Makefile.gnu	2019-06-19 12:40:29.704257159 +0200
+++ b/Makefile.gnu	2019-06-19 12:41:04.757002621 +0200
@@ -48,7 +48,6 @@
 
 dist: FreeImage
 	mkdir -p Dist
-	cp *.a Dist/
 	cp *.so Dist/
 	cp Source/FreeImage.h Dist/
 
@@ -55,7 +55,7 @@
 dos2unix:
 	@$(DOS2UNIX) $(SRCS) $(INCLS)
 
-FreeImage: $(STATICLIB) $(SHAREDLIB)
+FreeImage: $(SHAREDLIB)
 
 .c.o:
 	$(CC) $(CFLAGS) -c $< -o $@
@@ -71,9 +71,8 @@
 
 install:
 	install -d $(INCDIR) $(INSTALLDIR)
-	install -m 644 -o root -g root $(HEADER) $(INCDIR)
-	install -m 644 -o root -g root $(STATICLIB) $(INSTALLDIR)
-	install -m 755 -o root -g root $(SHAREDLIB) $(INSTALLDIR)
+	install -m 644 $(HEADER) $(INCDIR)
+	install -m 755 $(SHAREDLIB) $(INSTALLDIR)
 	ln -sf $(SHAREDLIB) $(INSTALLDIR)/$(VERLIBNAME)
 	ln -sf $(VERLIBNAME) $(INSTALLDIR)/$(LIBNAME)	
 #	ldconfig
