--- a/pcsx_rearmed_libretro.info	2019-04-05 03:50:25.993176768 +0200
+++ b/pcsx_rearmed_libretro.info	2019-04-05 03:50:34.479715474 +0200
@@ -1,6 +1,6 @@
 display_name = "Sony - PlayStation (PCSX ReARMed)"
 authors = "PCSX Team|notaz|Exophase"
-supported_extensions = "bin|cue|img|mdf|pbp|toc|cbn|m3u"
+supported_extensions = "bin|cue|img|mdf|pbp|toc|cbn|m3u|iso"
 corename = "PCSX ReARMed"
 manufacturer = "Sony"
 categories = "Emulator"
