commit 7e87ba4dcbb655c45ddd0047066a419e24de8b9a
Author: twinaphex <libretro@gmail.com>
Date:   Sat Sep 17 00:33:42 2016 +0200

    Dont use HAVE_GLES - turn it into HAVE_OPENGLES

diff --git b/menu/menu_displaylist.c a/menu/menu_displaylist.c
index df96b36..e01b84d 100644
--- b/menu/menu_displaylist.c
+++ a/menu/menu_displaylist.c
@@ -462,7 +462,7 @@ static uint64_t bytes_to_gb(uint64_t bytes)
 static int menu_displaylist_parse_system_info(menu_displaylist_info_t *info)
 {
    int controller;
-#if defined(HAVE_OPENGL) || defined(HAVE_OPENGLES)
+#if defined(HAVE_OPENGL) || defined(HAVE_GLES)
    gfx_ctx_ident_t ident_info;
 #endif
    char tmp[PATH_MAX_LENGTH]             = {0};
@@ -798,7 +798,7 @@ static int menu_displaylist_parse_system_info(menu_displaylist_info_t *info)
       }
    }
 
-#if defined(HAVE_OPENGL) || defined(HAVE_OPENGLES)
+#if defined(HAVE_OPENGL) || defined(HAVE_GLES)
    video_context_driver_get_ident(&ident_info);
    tmp_string = ident_info.ident;
 
commit e3c560014c6c616cccbb9d29ae565ce502991a20
Author: twinaphex <libretro@gmail.com>
Date:   Sat Sep 17 00:38:35 2016 +0200

    Turn HAVE_GLES into HAVE_OPENGLES

diff --git b/Makefile.common a/Makefile.common
index c005e80..586ec03 100644
--- b/Makefile.common
+++ a/Makefile.common
@@ -5,11 +5,11 @@ ifeq ($(HAVE_GL_CONTEXT),)
       HAVE_GL_CONTEXT=1
    endif
 
-   ifeq ($(HAVE_OPENGLES), 1)
+   ifeq ($(HAVE_GLES), 1)
       HAVE_GL_CONTEXT=1
    endif
 
-   ifeq ($(HAVE_OPENGLES3), 1)
+   ifeq ($(HAVE_GLES3), 1)
       HAVE_GL_CONTEXT=1
    endif
 endif
@@ -600,7 +600,7 @@ ifeq ($(HAVE_X11), 1)
 ifeq ($(HAVE_XCB),1)
    LIBS    += -lX11-xcb
 endif
-ifneq ($(HAVE_OPENGLES), 1)
+ifneq ($(HAVE_GLES), 1)
    OBJ += gfx/drivers_context/x_ctx.o
 endif
 endif
@@ -753,7 +753,7 @@ ifeq ($(HAVE_GL_CONTEXT), 1)
    endif
 ifeq ($(HAVE_FFMPEG), 1)
 ifneq ($(C89_BUILD), 1)
-ifneq ($(HAVE_OPENGLES), 1)
+ifneq ($(HAVE_GLES), 1)
    OBJ += cores/libretro-ffmpeg/fft/fft.o
    DEFINES += -Ideps -DHAVE_GL_FFT
    NEED_CXX_LINKER=1
@@ -761,10 +761,10 @@ endif
 endif
 endif
 
-   ifeq ($(HAVE_OPENGLES), 1)
+   ifeq ($(HAVE_GLES), 1)
       LIBS += $(GLES_LIBS)
       DEFINES += $(GLES_CFLAGS) -DHAVE_OPENGLES
-      ifeq ($(HAVE_OPENGLES3), 1)
+      ifeq ($(HAVE_GLES3), 1)
          DEFINES += -DHAVE_OPENGLES3
       else
          DEFINES += -DHAVE_OPENGLES2
diff --git b/Makefile.emscripten a/Makefile.emscripten
index d3efe60..7ff0221 100644
--- b/Makefile.emscripten
+++ a/Makefile.emscripten
@@ -9,7 +9,8 @@
 DEFINES += -DHAVE_OPENGL -DHAVE_OPENGLES -DHAVE_OPENGLES2 -DHAVE_EGL -DHAVE_OVERLAY -DHAVE_GLSL -DHAVE_FILTERS_BUILTIN 
 
 HAVE_EGL    = 1
-HAVE_OPENGLES = 1
+HAVE_OPENGL = 1
+HAVE_GLES = 1
 HAVE_RJPEG  = 0
 HAVE_RPNG   = 1
 HAVE_EMSCRIPTEN = 1
diff --git b/qb/config.libs.sh a/qb/config.libs.sh
index 9cd67b5..8a7b2fe 100644
--- b/qb/config.libs.sh
+++ a/qb/config.libs.sh
@@ -41,7 +41,7 @@ if [ "$HAVE_VIDEOCORE" = 'yes' ]; then
    [ -d /opt/vc/include ] && add_include_dirs /opt/vc/include
    [ -d /opt/vc/include/interface/vcos/pthreads ] && add_include_dirs /opt/vc/include/interface/vcos/pthreads
    [ -d /opt/vc/include/interface/vmcs_host/linux ] && add_include_dirs /opt/vc/include/interface/vmcs_host/linux
-   HAVE_OPENGLES='auto'
+   HAVE_GLES='auto'
    EXTRA_GL_LIBS="-lEGL -lGLESv2 -lbcm_host -lvcos -lvchiq_arm"
 fi
 
@@ -255,7 +255,7 @@ else
    HAVE_D3D9=no
 fi
 
-if [ "$HAVE_OPENGL" != 'no' ] && [ "$HAVE_OPENGLES" != 'yes' ]; then
+if [ "$HAVE_OPENGL" != 'no' ] && [ "$HAVE_GLES" != 'yes' ]; then
    if [ "$OS" = 'Darwin' ]; then
       check_header OPENGL "OpenGL/gl.h"
       check_lib OPENGL "-framework OpenGL"
@@ -338,14 +338,14 @@ fi
 check_pkgconf LIBXML2 libxml-2.0
 
 if [ "$HAVE_EGL" = "yes" ]; then
-   if [ "$HAVE_OPENGLES" != "no" ]; then
+   if [ "$HAVE_GLES" != "no" ]; then
       if [ "$GLES_LIBS" ] || [ "$GLES_CFLAGS" ]; then
          echo "Notice: Using custom OpenGLES CFLAGS ($GLES_CFLAGS) and LDFLAGS ($GLES_LIBS)."
          add_define_make GLES_LIBS "$GLES_LIBS"
          add_define_make GLES_CFLAGS "$GLES_CFLAGS"
       else
-         HAVE_OPENGLES=auto check_pkgconf GLES glesv2
-         [ "$HAVE_OPENGLES" = "no" ] && HAVE_OPENGLES=auto check_lib GLES "-lGLESv2 $EXTRA_GL_LIBS" && add_define_make GLES_LIBS "-lGLESv2 $EXTRA_GL_LIBS"
+         HAVE_GLES=auto check_pkgconf GLES glesv2
+         [ "$HAVE_GLES" = "no" ] && HAVE_GLES=auto check_lib GLES "-lGLESv2 $EXTRA_GL_LIBS" && add_define_make GLES_LIBS "-lGLESv2 $EXTRA_GL_LIBS"
       fi
    fi
    if [ "$HAVE_VG" != "no" ]; then
@@ -357,7 +357,7 @@ if [ "$HAVE_EGL" = "yes" ]; then
    fi
 else
    HAVE_VG=no
-   HAVE_OPENGLES=no
+   HAVE_GLES=no
 fi
 
 check_pkgconf V4L2 libv4l2
@@ -367,7 +367,7 @@ if [ "$OS" = 'Darwin' ]; then
 elif [ "$OS" = 'Win32' ]; then
    HAVE_FBO=yes
 else
-   if [ "$HAVE_OPENGLES" = "yes" ]; then
+   if [ "$HAVE_GLES" = "yes" ]; then
       [ $HAVE_FBO != "no" ] && HAVE_FBO=yes
    else
       check_lib FBO -lGL glFramebufferTexture2D
@@ -434,7 +434,7 @@ if [ "$HAVE_MATERIALUI" != 'no' ] || [ "$HAVE_XMB" != 'no' ] || [ "$HAVE_ZARCH"
 		HAVE_XMB=no
       HAVE_ZARCH=no
 		echo "Notice: RGUI not available, MaterialUI, XMB and ZARCH will also be disabled."
-	elif [ "$HAVE_OPENGL" = 'no' ] && [ "$HAVE_OPENGLES" = 'no' ] && [ "$HAVE_VULKAN" = 'no' ]; then
+	elif [ "$HAVE_OPENGL" = 'no' ] && [ "$HAVE_GLES" = 'no' ] && [ "$HAVE_VULKAN" = 'no' ]; then
 		HAVE_MATERIALUI=no
 		HAVE_XMB=no
       HAVE_ZARCH=no
diff --git b/qb/config.params.sh a/qb/config.params.sh
index 4e57866..b510dec 100644
--- b/qb/config.params.sh
+++ a/qb/config.params.sh
@@ -17,8 +17,8 @@ HAVE_LIBRETRO=             # Libretro library used
 HAVE_ASSETS_DIR=           # Assets install directory
 HAVE_BIN_DIR=              # Binary install directory
 HAVE_MAN_DIR=              # Manpage install directory
-HAVE_OPENGLES_LIBS=        # Link flags for custom GLES library
-HAVE_OPENGLES_CFLAGS=      # C-flags for custom GLES library
+HAVE_GLES_LIBS=            # Link flags for custom GLES library
+HAVE_GLES_CFLAGS=          # C-flags for custom GLES library
 HAVE_THREADS=auto          # Threading support
 HAVE_FFMPEG=auto           # FFmpeg recording support
 C89_FFMPEG=no
@@ -30,11 +30,11 @@ C89_NETWORKGAMEPAD=no
 HAVE_NETPLAY=auto          # Netplay support (requires networking)
 HAVE_D3D9=yes              # Direct3D 9 support
 HAVE_OPENGL=auto           # OpenGL support
+HAVE_GLES=no               # Use GLESv2 instead of desktop GL
 HAVE_MALI_FBDEV=no         # Mali fbdev context support
 HAVE_VIVANTE_FBDEV=no      # Vivante fbdev context support
 HAVE_OPENDINGUX_FBDEV=no   # Opendingux fbdev context support
-HAVE_OPENGLES=no           # Use GLESv2 instead of desktop GL
-HAVE_OPENGLES3=no          # OpenGLES3 support
+HAVE_GLES3=no              # OpenGLES3 support
 HAVE_X11=auto              # everything X11.
 HAVE_OMAP=no               # OMAP video support
 HAVE_XINERAMA=auto         # Xinerama support.
