--- a/qb/config.libs.sh 2016-06-30 11:07:32.883379420 +0200
+++ b/qb/config.libs.sh	2016-06-30 11:07:39.543379378 +0200
@@ -395,7 +395,7 @@
 fi
 check_lib STRCASESTR "$CLIB" strcasestr
 check_lib MMAP "$CLIB" mmap
-check_lib VULKAN -lvulkan vkCreateInstance
+check_lib VULKAN -lvulkan_intel vkCreateInstance
 
 if [ "$HAVE_VULKAN" != 'no' ] && [ ! -e deps/glslang/glslang/README.md ]; then
 	echo "Warning: glslang submodule not loaded, can't use Vulkan."
