--- a/CMakeLists.txt	2019-05-28 12:55:38.651046310 +0200
+++ b/CMakeLists.txt	2019-05-28 12:55:31.404475373 +0200
@@ -21,7 +21,7 @@
 set(SDL_TTF_VERSION "${SDL_TTF_MAJOR_VERSION}.${SDL_TTF_MINOR_VERSION}.${SDL_TTF_MICRO_VERSION}")
 
 ##### library generation #####
-add_library(SDL2_ttf SDL_ttf.c SDL_ttf.h)
+add_library(SDL2_ttf SHARED SDL_ttf.c SDL_ttf.h)
 target_link_libraries(SDL2_ttf SDL2::SDL2 Freetype::Freetype)
 target_include_directories(SDL2_ttf PUBLIC $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/SDL2>)
 
