--- a/networking/wget.c	2017-06-21 22:46:45.778029489 +0200
+++ b/networking/wget.c	2017-06-21 22:46:56.081475064 +0200
@@ -1093,12 +1093,6 @@
 		}
 
 		fflush(sfp);
-		/* If we use SSL helper, keeping our end of the socket open for writing
-		 * makes our end (i.e. the same fd!) readable (EAGAIN instead of EOF)
-		 * even after child closes its copy of the fd.
-		 * This helps:
-		 */
-		shutdown(fileno(sfp), SHUT_WR);
 
 		/*
 		 * Retrieve HTTP response line and check for "200" status code.
