--- a/es-app/src/guis/GuiMenu.cpp	2019-09-16 23:22:50.800740004 +0200
+++ b/es-app/src/guis/GuiMenu.cpp	2019-09-16 23:29:52.832571287 +0200
@@ -488,16 +488,44 @@
 
 		if(Settings::getInstance()->getBool("ShowExit"))
 		{
-			row.elements.clear();
-			row.makeAcceptInputHandler([window] {
-				window->pushGui(new GuiMsgBox(window, "REALLY QUIT?", "YES",
-					[] {
-					Scripting::fireEvent("quit");
-					quitES();
+		row.elements.clear();
+		row.makeAcceptInputHandler([window] {
+			window->pushGui(new GuiMsgBox(window, "QUIT TO RETROARCH?", "YES",
+				[] {
+				runSystemCommand("systemd-run /usr/bin/es-retroarch.sh");
+				SDL_Event ev;
+				ev.type = SDL_QUIT;
+				SDL_PushEvent(&ev);
+			}, "NO", nullptr));
+		});
+		row.addElement(std::make_shared<TextComponent>(window, "START RETROARCH", Font::get(FONT_SIZE_MEDIUM), 0x777777FF), true);
+		s->addRow(row);
+
+		row.elements.clear();
+		row.makeAcceptInputHandler([window] {
+			window->pushGui(new GuiMsgBox(window, "QUIT TO RPCS3?", "YES",
+				[] {
+				runSystemCommand("systemd-run /usr/bin/es-rpcs3.sh");
+				SDL_Event ev;
+				ev.type = SDL_QUIT;
+				SDL_PushEvent(&ev);
+				}, "NO", nullptr));
+		});
+		row.addElement(std::make_shared<TextComponent>(window, "START RPCS3", Font::get(FONT_SIZE_MEDIUM), 0x777777FF), true);
+		s->addRow(row);
+
+		row.elements.clear();
+		row.makeAcceptInputHandler([window] {
+			window->pushGui(new GuiMsgBox(window, "QUIT TO TERMINAL?", "YES",
+				[] {
+				runSystemCommand("systemd-run /usr/bin/es-terminal.sh");
+				SDL_Event ev;
+				ev.type = SDL_QUIT;
+				SDL_PushEvent(&ev);
 				}, "NO", nullptr));
-			});
-			row.addElement(std::make_shared<TextComponent>(window, "QUIT EMULATIONSTATION", Font::get(FONT_SIZE_MEDIUM), 0x777777FF), true);
-			s->addRow(row);
+		});
+		row.addElement(std::make_shared<TextComponent>(window, "START TERMINAL", Font::get(FONT_SIZE_MEDIUM), 0x777777FF), true);
+		s->addRow(row);
 		}
 	}
 	row.elements.clear();
