--- a/source3/nmbd/nmbd_namequery.c	2019-08-26 14:16:57.328428742 +0200
+++ b/source3/nmbd/nmbd_namequery.c	2019-08-26 14:19:14.385094539 +0200
@@ -104,26 +104,8 @@
 				before sending out a query. This is a task for another day, though.
 			*/
 		}
-	} else if( rrec->num_msgs > 1) {
+	}  
 
-		if( DEBUGLVL( 0 ) ) {
-			if (nmb->answers)
-				putip( (char *)&answer_ip, &nmb->answers->rdata[2] );
-			dbgtext( "query_name_response: " );
-			dbgtext( "Multiple (%d) responses ", rrec->num_msgs );
-			dbgtext( "received for a query on subnet %s ", subrec->subnet_name );
-			dbgtext( "for name %s.\nThis response ", nmb_namestr(question_name) );
-			dbgtext( "was from IP %s, reporting ", inet_ntoa(p->ip) );
-			dbgtext( "an IP address of %s.\n", inet_ntoa(answer_ip) );
-		}
-
-		/* We have already called the success or fail function, so we
-			don't call again here. Leave the response record around in
-			case we get more responses. */
-
-		return; 
-	}
-  
 	if(success && rrec->success_fn)
 		(*(query_name_success_function)rrec->success_fn)(subrec, rrec->userdata, question_name, answer_ip, nmb->answers);
 	else if( rrec->fail_fn)
