--- a/es-app/src/guis/GuiMenu.cpp	2019-04-10 15:28:24.562743653 +0200
+++ b/es-app/src/guis/GuiMenu.cpp	2019-04-10 15:29:11.952110236 +0200
@@ -474,6 +474,15 @@
 	if (UIModeController::getInstance()->isUIModeFull())
 	{
 		row.makeAcceptInputHandler([window] {
+			window->pushGui(new GuiMsgBox(window, "QUIT TO KODI?", "YES",
+				[] {
+				runSystemCommand("systemd-run /usr/bin/es-kodi.sh");
+				SDL_Event ev;
+				ev.type = SDL_QUIT;
+				SDL_PushEvent(&ev);
+			}, "NO", nullptr));
+		});
+		row.makeAcceptInputHandler([window] {
 			window->pushGui(new GuiMsgBox(window, "REALLY RESTART?", "YES",
 				[] {
 				Scripting::fireEvent("quit");
