--- a/backends/platform/libretro/build/Makefile	2016-08-12 13:25:27.666619492 +0200
+++ b/backends/platform/libretro/build/Makefile	2016-08-12 13:27:55.641126030 +0200
@@ -212,7 +212,7 @@
 endif
 
 POSIX = 1
-DEFINES += -DFS_TYPE_POSIX -DUSE_VORBIS -DUSE_TREMOR -DUSE_ZLIB 
+DEFINES += -DFS_TYPE_POSIX -DUSE_VORBIS -DUSE_TREMOR -DUSE_ZLIB -DUSE_MAD -DUSE_FLAC
 MODULE_DIRS += $(retrodir)
 
 BACKEND := libretro
